Makefile

```
@@ -9,102 +8,85 @@ NUGET_PKG_NAME   := Pulumi.Kubernetes
-
-ifeq ($(strip ${DOTNET_SUFFIX}),)
-	DOTNET_VERSION := $(strip ${DOTNET_PREFIX})
-else
-	DOTNET_VERSION := $(strip ${DOTNET_PREFIX})-$(strip ${DOTNET_SUFFIX})
-endif
-
+WORKING_DIR     := $(shell pwd)
 TESTPARALLELISM := 4
 
-$(OPENAPI_FILE)::
+openapi_file::
 	@mkdir -p $(OPENAPI_DIR)
-	test -f $(OPENAPI_FILE) || $(CURL) -s -L $(SWAGGER_URL) > $(OPENAPI_FILE)
+	test -f $(OPENAPI_FILE) || curl -s -L $(SWAGGER_URL) > $(OPENAPI_FILE)
 
 k8sgen::
-	cd provider && $(GO) install $(VERSION_FLAGS) $(PROJECT)/provider/v2/cmd/$(CODEGEN)
+	(cd provider && go build -a -o $(WORKING_DIR)/bin/${CODEGEN} -ldflags "-X ${PROJECT}/${VERSION_PATH}=${VERSION}" ${PROJECT}/${PROVIDER_PATH}/cmd/$(CODEGEN))
 
-$(SCHEMA_FILE):: k8sgen $(OPENAPI_FILE)
-	$(call STEP_MESSAGE)
+schema::
 	@echo "Generating Pulumi schema..."
-	$(CODEGEN) schema $(OPENAPI_FILE) $(CURDIR)
+	$(WORKING_DIR)/bin/${CODEGEN} schema $(OPENAPI_FILE) $(CURDIR)
 	@echo "Finished generating schema."
 
-k8sprovider:: $(SCHEMA_FILE)
-	$(CODEGEN) kinds $(SCHEMA_FILE) $(CURDIR)
-	cd provider && VERSION=${VERSION} $(GO) generate cmd/${PROVIDER}/main.go
-	cd provider && $(GO) install $(VERSION_FLAGS) $(PROJECT)/provider/v2/cmd/$(PROVIDER)
+k8sprovider::
+	$(WORKING_DIR)/bin/${CODEGEN} kinds $(SCHEMA_FILE) $(CURDIR)
+	(cd provider && VERSION=${VERSION} go generate cmd/${PROVIDER}/main.go)
+	(cd provider && go build -a -o $(WORKING_DIR)/bin/${PROVIDER} -ldflags "-X ${PROJECT}/${VERSION_PATH}=${VERSION}" $(PROJECT)/provider/v2/cmd/$(PROVIDER))
+
+test_provider::
+	cd provider/pkg && go test -short -v -count=1 -cover -timeout 2h -parallel ${TESTPARALLELISM} ./...
 
-dotnet_sdk:: k8sgen $(OPENAPI_FILE)
-	$(CODEGEN) -version=${VERSION} dotnet $(SCHEMA_FILE) $(CURDIR)
+dotnet_sdk:: DOTNET_VERSION := $(shell pulumictl get version --language dotnet)
+dotnet_sdk::
+	$(WORKING_DIR)/bin/$(CODEGEN) -version=${DOTNET_VERSION} dotnet $(SCHEMA_FILE) $(CURDIR)
 	rm -rf sdk/dotnet/bin/Debug
 	cd ${PACKDIR}/dotnet/&& \
-		echo "${VERSION:v%=%}" >version.txt && \
+		echo "${DOTNET_VERSION}" >version.txt && \
 		dotnet build /p:Version=${DOTNET_VERSION}
 
-go_sdk:: k8sgen $(SCHEMA_FILE)
-	$(CODEGEN) -version=${VERSION} go $(SCHEMA_FILE) $(CURDIR)
+go_sdk::
+	$(WORKING_DIR)/bin/$(CODEGEN) -version=${VERSION} go $(SCHEMA_FILE) $(CURDIR)
 
-nodejs_sdk:: k8sgen $(SCHEMA_FILE)
-	$(CODEGEN) -version=${VERSION} nodejs $(SCHEMA_FILE) $(CURDIR)
+nodejs_sdk:: VERSION := $(shell pulumictl get version --language javascript)
+nodejs_sdk::
+	$(WORKING_DIR)/bin/$(CODEGEN) -version=${VERSION} nodejs $(SCHEMA_FILE) $(CURDIR)
 	cd ${PACKDIR}/nodejs/ && \
 		yarn install && \
 		yarn run tsc
 	cp README.md LICENSE ${PACKDIR}/nodejs/package.json ${PACKDIR}/nodejs/yarn.lock ${PACKDIR}/nodejs/bin/
 	sed -i.bak 's/$${VERSION}/$(VERSION)/g' ${PACKDIR}/nodejs/bin/package.json
 
-python_sdk:: k8sgen $(SCHEMA_FILE)
+nodejs_sdk:: PYPI_VERSION := $(shell pulumictl get version --language python)
```

I think this should be `python_sdk`
```suggestion
python_sdk:: PYPI_VERSION := $(shell pulumictl get version --language python)
```